src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -635,7 +648,7 @@ def prepare_inputs_for_generation(self, input_ids, past=None, attention_mask=Non
             attention_mask = input_ids.new_ones(input_shape)
 
         # cut decoder_input_ids if past is used
-        if past is not None:
+        if past is not None and past[0] is not None:
```

Comment:
Out of curiosity why is the second statement needed here? The `past[0] is not None` part?